twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -367,6 +418,15 @@ scrooge_aspect = aspect(
                 Label(
                     "//external:io_bazel_rules_scala/dependency/thrift/util_core",
                 ),
+                Label(
```

Comment:
I'm a little concerned that everyone now needs to depend on these jars even though many won't need them. I wish we could find some way to only put these jars on the classpath if the thrifts have services.

Maybe we could add `has_services` to the thrift rules, and then only for thrifts that claim to have services to we include these on the classpath. What do you think?

Increasing the classpath also slows down compilation and invalidates caches in some cases, so minimizing is a good idea.

Reply:
Yeah, I was concerned about that too. I like the idea of `has_services`, but I'm not sure how to cleanly handle the failure case. If `has_services` isn't set, then the user will get dependency errors and it's not quite obvious why.

Would it be possible to somehow catch that specific failure case? We could also just print out a message saying "If you have services, this might be due to `has_services = False`"

Reply:
I think if the transitive dependencies don't include services, we just wouldn't generate the service output, so they wouldn't get dependency errors. If they do transitively have services, then bazel would error since scalac could not find required code when compiling the services.

I think it should be okay since the scrooge compiler can be instructed not to generate the services. Does that sound like it would work?

Reply:
@johnynek Finally getting back to this. I see in the comments in `_compile_to_scala` there was mention of putting `include_services` on `thrift_info`. I don't fully grok providers and aspects yet, but it seems like this would make more sense to put on the `scrooge_scala_library` rule, since if it's put on `ThriftInfo`, a single `thrift_library` that wants services would cause all `scrooge_scala_library` targets to generate services instead of letting the user of `scrooge_scala_library` decide.

Will need to spend some time reading up on providers and aspects to understand this code better.